src/main/resources/assets/js/components/AppHealthComponent.jsx

```diff
@@ -0,0 +1,43 @@
+/** @jsx React.DOM */
+
+"use strict";
+
+var React = require("react/addons");
+
+module.exports = React.createClass({
+    propTypes: {
+      data : React.PropTypes.array.isRequired,
+      className : React.PropTypes.string
+    },
+
+    getDefaultProps: function () {
+      return {
+      };
+    },
+
+    render: function () {
+      // normalize data to add up to 100%
+      var dataSum = this.props.data.reduce(function (a, x) {
+        return a + x.quantity;
+      }, 0);
+      var normalizedData = this.props.data.map(function (d) {
+        return {
+          quantity: d.quantity * 100 / dataSum,
+          className: ("progress-bar " + (d.className || "")).trim()
```

Comment:
Maybe we could do this a little differently using React class sets, and get some space in there for readability?
```
var normalizedData = data.map(function (d) {
  var nodeClasses = {
    "progress-bar": true
  };

  if (d.className != null) {
    nodeClasses[d.className] = true;
  }

  return {
    quantity: d.quantity * 100 / dataSum,
    className: React.addons.classSet(nodeClasses)
  };
});

var classes = {
  "progress": true
};

if (this.props.className != null) {
  classes[this.props.className] = true;
}

var classSet = React.addons.classSet(classes);
...
return (
  <div className={classSet}>
```